src/lib.rs

```diff
@@ -8,37 +8,36 @@
 //! ```rust,no_run
 //! # extern crate dkregistry;
 //! # extern crate tokio;
-//! # fn main() {
-//! # fn run() -> dkregistry::errors::Result<()> {
+//! # #[tokio::main]
+//! # async fn main() {
+//! # async fn run() -> dkregistry::errors::Result<()> {
 //! #
-//! use tokio::runtime::current_thread::Runtime;
 //! use dkregistry::v2::Client;
 //!
 //! // Check whether a registry supports API v2.
 //! let host = "quay.io";
-//! let mut runtime = Runtime::new()?;
 //! let dclient = Client::configure()
 //!                      .insecure_registry(false)
 //!                      .registry(host)
 //!                      .build()?;
-//! let check = dclient.is_v2_supported();
-//! match runtime.block_on(check)? {
+//! match dclient.is_v2_supported().await? {
 //!     false => println!("{} does NOT support v2", host),
 //!     true => println!("{} supports v2", host),
 //! };
 //! #
 //! # Ok(())
 //! # };
-//! # run().unwrap();
+//! # run().await.unwrap();
 //! # }
 //! ```
 
+#![feature(async_closure)]
```

Comment:
This is an unstable feature, which doesn't work on stable toolchains. Please drop it.

Reply:
For reference: https://github.com/rust-lang/rust/issues/62290

Reply:
done